src/platform/intel/cavs/include/cavs/dw-dma.h

```diff
@@ -45,8 +45,8 @@
 #define DW_CFG_CTL_HI_UPD_EN	BIT(5)
 
 /* CFG_HI */
-#define DW_CFGH_DST_PER_EXT(x)		SET_BITS(31, 30, x)
-#define DW_CFGH_SRC_PER_EXT(x)		SET_BITS(29, 28, x)
+#define DW_CFGH_DST_PER_EXT(x)		SET_BITS(31, 26, x)
```

Comment:
Why do the overlap? It doesn't seem correct.

Reply:
they don't overlap. x above has its low 4 bits = 0. So it's 0 or in the range `0x10 <= x <= 0x30` When you shift it by 24 bits you get bits 27..24 unaffected and bits 29..28 set. When you shift it by 26 bits you get bits 29..26 unaffected and only bits 31..30 set.

Reply:
But doing it this way makes it look weird and not right. Register definitions should be aligned with the real hardware. Better way would be to change `DW_CFGH_DST` and `DW_CFGH_SRC` macros:
`(DW_CFGH_DST_PER_EXT(((x) & 0x30) >> 4) | DW_CFGH_DST_PER((x) & 0xF))`

Reply:
I agreed with @tlauda 
If the only valid bits is 31,30, we should change the X value but not the definition here.
If we need some X value with low 4 bit is 0, you should add some comments here to explain.

Reply:
@tlauda sure, I agree, that this way the code is working around the definition. But it was your patch to begin using SET_BITS() here, so, I didn't want to change that. As for your proposed new definition, it's equivalent to mine, but it adds redundancy. It first shifts right by four bits to then shift left by 30 bits, which is in this case equivalent to just shifting right by 26 bits, which is exactly what my patch does. We can add a comment as @xiulipan has suggested, sure.

Reply:
@lyakh Yes, it's redundant, but at least the definition is aligned with the hardware register definition. It doesn't seem right to modify it just to avoid redundancy.

Reply:
@lyakh @tlauda 
Then let make the definition here map to hardware and add comment or change the logic in `DW_CFGH_DST` and `DW_CFGH_SRC`

Reply:
I think I've got a better version of the patch now.